headerCheck/factory/nodesCoordinator.go

```diff
@@ -116,15 +116,19 @@ func createArgsNodesShuffler(
 		maxNodesChangeConfigs = append(maxNodesChangeConfigs, maxNodesChangeConfig)
 	}
 
+	enableEpochsHandler, err := enablers.NewEnableEpochsHandler(eec.EnableEpochs, &disabled.EpochNotifier{})
+	if err != nil {
```

Comment:
use this and remove the enable epochs handler field from core components, or instantiate it there

Reply:
:+1:

Reply:
we need enable epochs handler in nodes coordinator also, so I suggest to instantiate it separately, maybe in core components as initially intended